lib/pat.c

```diff
@@ -146,10 +146,29 @@ pat_node_get_right(grn_pat *pat, pat_node_common *node)
   }
 }
 
-#define PAT_DEL(x)     ((x)->bits & PAT_DELETING)
-#define PAT_IMD(x)     ((x)->bits & PAT_IMMEDIATE)
-#define PAT_LEN(x)     (uint32_t)(((x)->bits >> 3) + 1)
-#define PAT_CHK(x)     ((x)->check)
+static inline grn_id
+pat_node_get_left(grn_pat *pat, pat_node_common *node)
+{
+  if (pat_is_key_large(pat)) {
+    return node->node_large.lr[0];
+  } else {
+    return node->node.lr[0];
+  }
+}
+
+#define PAT_DEL(x) ((x)->bits & PAT_DELETING)
+#define PAT_IMD(x) ((x)->bits & PAT_IMMEDIATE)
+#define PAT_LEN(x) (uint32_t)(((x)->bits >> 3) + 1)
+#define PAT_CHK(x) ((x)->check)
+static inline uint16_t
+pat_chk(grn_pat *pat, pat_node_common *node)
```

Comment:
If it is the same naming rule as `pat_node_get_left()`, `pat_node_get_check()` seems better?

Reply:
I see.
I changed to `pat_node_get_check()`.